src/OpenLoco/src/World/Station.cpp

```diff
@@ -873,6 +880,186 @@ namespace OpenLoco
+                    break;
+                }
+                case StationType::roadStation:
+                {
+                    // Loop from start as road element can be variable number of elements below station
+                    auto tile = TileManager::get(pos);
+                    for (auto& el2 : tile)
+                    {
+                        if (&el2 == reinterpret_cast<TileElement*>(&elStation))
+                        {
+                            break;
+                        }
+                        auto* elRoad = el2.as<RoadElement>();
+                        if (elRoad == nullptr)
+                        {
+                            continue;
+                        }
+                        if (elRoad->baseZ() != elStation->baseZ())
+                        {
+                            continue;
+                        }
+                        if (elRoad->isAiAllocated() || elRoad->isGhost())
+                        {
+                            continue;
+                        }
+                        auto* roadObj = ObjectManager::get<RoadObject>(elRoad->roadObjectId());
+                        station->flags |= roadObj->hasFlags(RoadObjectFlags::unk_01)
+                            ? StationFlags::transportModeRail
+                            : StationFlags::transportModeRoad;
+                    }
+                    auto* roadStationObj = ObjectManager::get<RoadStationObject>(elStation->objectId());
+                    if (roadStationObj->hasFlags(RoadStationFlags::passenger))
+                    {
+                        acceptedCargoTypes |= 1U << roadStationObj->cargoType;
+                    }
+                    else if (roadStationObj->hasFlags(RoadStationFlags::freight))
+                    {
+                        // Exclude passengers from this station type
+                        acceptedCargoTypes |= 0xFFFFFFFFU & ~(1U << roadStationObj->cargoType);
+                    }
+                    else
+                    {
+                        acceptedCargoTypes = 0xFFFFFFFFU;
+                    }
+                    break;
+                }
+                case StationType::airport:
+                    station->flags |= StationFlags::transportModeAir;
+                    acceptedCargoTypes = 0xFFFFFFFFU;
+                    break;
+                case StationType::docks:
+                    station->flags |= StationFlags::transportModeWater;
+                    acceptedCargoTypes = 0xFFFFFFFFU;
+                    break;
+            }
+        }
+
+        for (auto cargoType = 0U; cargoType < ObjectManager::getMaxObjects(ObjectType::cargo); ++cargoType)
+        {
+            auto* cargoObj = ObjectManager::get<CargoObject>(cargoType);
+            if (cargoObj == nullptr)
+            {
+                continue;
+            }
+            station->cargoStats[cargoType].flags &= ~StationCargoStatsFlags::flag1;
```

Comment:
There is some code v. similar to this in the station catchment area calcs. Might be an idea to look at extracting it out.